### PR TITLE
TIP-694: Add completeness integration tests

### PR DESCRIPTION
**Pull request by SamirBoulil**
(we'll squash commits before merging)

**Description (for Contributor and Core Developer)**

Add integration tests regarding the Completeness feature, it takes into account how the completeness is calculated given:
- the available attribute types in the PIM
- the localizable specificity of an attribute (as well as locale specific)
- the scopable specificity of an attribute
- the locale/scope specificities mixed
- the non required attributes do not influence the calculation

The goal is to be able to refactor the completeness without any fear of breaking something.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        $this->assertEquals($requiredCount, $completeness->getRequiredCount());
+        $this->assertEquals(1, $completeness->getMissingCount());
+    }
+
+    /**
+     * @param ProductInterface $product
+     * @param string           $localeCode
+     * @param int              $requiredCount
+     */
+    private function assertComplete(ProductInterface $product, $localeCode, $requiredCount)
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php]: Drawing your attention to this last parameter

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/AttributeType/NumberAttributeTypeCompletenessIntegration.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        $this->assertComplete($productComplete);
+    }
+
+    public function testCompleteWithZero()
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/AttributeType/NumberAttributeTypeCompletenessIntegration.php]: Should be done in `testComplete`

**Comment by SamirBoulil** [comment]
Can't GTM it. but GTM

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+/**
+ * Checks that the completeness has been well calculated for localisable and locale specific attribute types.
+ *
+ * We test from the minimal catalog that contains only 1 channel. The locales fr_FR and en_US are activated.
+ *
+ * The completeness calculation is tested for:
+ *      - 1 localisable attribute
+ *      - 1 locale specific attribute
+ *
+ * For each test, the we create a family where the attribute is required.
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php]: For each test, we create

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForScopableAttributeIntegration.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+use Pim\Component\Catalog\AttributeTypes;
+use Pim\Component\Catalog\Model\ProductInterface;
+
+/**
+ * Checks that the completeness has been well calculated for scopable attribute types.
+ *
+ * We test from the minimal catalog that contains only one channel, with one locale activated.
+ *
+ * For each test, the we create a family where the scopable attribute is required.
```
Comment (damien-carcel) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForScopableAttributeIntegration.php]: For each test, we create

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/AttributeType/NumberAttributeTypeCompletenessIntegration.php (hunk 2 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/AttributeType/NumberAttributeTypeCompletenessIntegration.php]: done

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForScopableAttributeIntegration.php (hunk 4 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForScopableAttributeIntegration.php]: done

File: src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php (hunk 3 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/tests/integration/Completeness/CompletenessForLocalisableAttributeIntegration.php]: done

**Comment by SamirBoulil** [comment]
GTM

**Pull request merged**